src/components/navigation/navigation-item/navigation-item.component.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+  ContentChildren,
+  QueryList,
+  Renderer2,
+  ElementRef,
+  Optional,
+  SkipSelf,
+  AfterContentInit,
+  ContentChild,
+  Output,
+  AfterViewInit,
+  HostListener
+} from '@angular/core';
+import { Router, RouterLinkActive, NavigationEnd, RouterState, ActivatedRoute, UrlTree } from '@angular/router';
+import 'rxjs/add/operator/filter';
+
+@Component({
+    selector: '[ux-navigation-item]',
+    templateUrl: './navigation-item.component.html',
+    host: {
+        '[class.active]': 'active',
+        '[class.selected]': 'expanded',
+    },
+    encapsulation: ViewEncapsulation.None
+})
+export class NavigationItemComponent implements AfterViewInit {
+    @Input() header: string;
+    @Input() icon: string;
+    @Input() expanded: boolean = false;
+    @Input() link?: string;
+
+    @Input()
+    set level(value: number) {
+      this._level = value;
+    }
+
+    get level(): number {
+      return this._level;
+    }
+
+    @Output()
+    get active() {
+      if (this.link) {
+          return this._router.isActive(this.link, true);
+      } else {
+          return this._active;

```

Comment:
I feel like we had this discussion before - what is the use case for a nav item being active other than being the active route?

Reply:
get active() modified - only active routes can return true.